### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by mmorhun**
Adds more detailed information about the extension into readme file.
Explains purpose for this extension, how it works, how to use and test it.

Resolves https://github.com/eclipse/che-theia-hosted-plugin-manager-extension/issues/8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,32 @@
...
+If Theia runs natively a user can just use direct url, but in case when Theia is running inside Eclipse Che workspace direct url won't work. There should be a dedicated server in workspace config to expose needed port and this server has a preview url which is designed to access the resource from outside.
+
+#### How does this extension work?
+
+It retrieves hosted instance url form workspace config.
+Supposed, that hosted instance server has attribute "type" with "ide-dev" value.
+
+#### How to test the extension
+
+Just start hosted plugin instance (use `Hosted Plugin: Start Instance` command). If a new browser tab with hosted instance is opened then it works as expected.
```
Comment (sunix) [inline README.md]: Maybe provide a link to a sample plugin to be used in Github or how to create it quickly.

**Comment by sunix** [comment]
Looks good. Last question:
- Do we have any documentation about features/user stories that hosted mode support ?
- Would it be part of this extension README file ? or should it be someone else ? if it is somewhere else, we should have a link to this here.

**Comment by mmorhun** [comment]
> Do we have any documentation about features/user stories that hosted mode support ?

No, we haven't

> Would it be part of this extension README file ? or should it be someone else ? if it is somewhere else, we should have a link to this here.

It definitely should not be here. I think it should be in Theia / Eclipse Che docs.

> A kind of question I have without such documentation could be 'does it support hot reload ?'

Without documentation there could be a lot questions.
No, it doesn't support hot reload.

**Comment by sunix** [comment]
FYI https://github.com/eclipse/che/issues/11004

**Pull request merged**